.github/workflows/integration.yml

```diff
@@ -0,0 +1,36 @@
+name: Integration Test
+on: [push]
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.4.0
+      - uses: actions/setup-go@v2
+      - name: build
+        run: go build -o main ./cmd/
+      - name: Upload build
+        uses: 'actions/upload-artifact@v2'
+        with:
+          name: main
+          path: main
+  provenance:
+    runs-on: ubuntu-latest
+    steps:
+      # Download build from previous job
+      - uses: 'actions/download-artifact@v2'
```

Comment:
how about moving this into the provenance action? This way usage becomes:
```
uses: slsa-provenance@
  with
     - pathto-exe
```

digest can be computed by the action, checkout can be hidden as well if we only support the default branch.
Something we do't know yet is how to retrieve the correct binary from the registry?

Reply:
done, but i didnt see how local dockerfile could be compatible with composite, so i had to make an intermediate action.yaml

Reply:
I don't think dockerfiles are compatible with composite. But that's ok since we can just build everything within a `run/step`, right?

Reply:
Yeah, I think my alternative here was to use a run to invoke docker, ill iterate on this when i've sorted out any other problems (e.g. github token)